src/wsproxy.ml

```diff
@@ -45,7 +46,15 @@ let start path handler =
          Lwt_unix.close fd_sock2 >>= fun _ ->
          List.iter (fun fd -> Printf.printf "got fd: %d\n%!" (Obj.magic fd)) newfds;
          Printf.printf "About to fixup the fd\n%!";
-         ignore(handler (Lwt_unix.of_unix_file_descr (List.hd newfds)) msg);
+         Lwt.catch
+          (fun () ->
+            ignore(handler (Lwt_unix.of_unix_file_descr (List.hd newfds)) msg);
+            Lwt.return ()
+          )
```

Comment:
It seems to me that, due to the use of `Lwt.catch` instead of `Lwt.finalize`, we only close all of the file descriptors in `newfds` when an exception occurs. When no exception happens, our handler does close the file descriptors we pass to it, but we only pass to it the first one from `newfds`, and the others will remain unclosed. Or are we sure that we will always have only one fd in `newfds`?